packages/auth/amplify_auth_cognito_dart/lib/src/sdk/sdk_exception.dart

```diff
@@ -27,31 +27,27 @@ sealed class CognitoServiceException extends core.AuthServiceException {
 /// {@endtemplate}
 final class LambdaException extends CognitoServiceException {
   /// {@macro amplify_auth_cognito_dart.sdk.lambda_exception}
-  factory LambdaException(
-    String message, {
-    String? recoverySuggestion,
-    Object? underlyingException,
-  }) {
-    final match = _errorRegex.firstMatch(message);
-    final lambdaName = match?.group(1);
-    final parsedMessage = match?.group(2);
-    if (parsedMessage != null) {
-      message = parsedMessage;
-    }
-    return LambdaException._(
-      message,
-      lambdaName: lambdaName,
-      recoverySuggestion: recoverySuggestion,
-      underlyingException: underlyingException,
-    );
-  }
-
-  const LambdaException._(
+  const LambdaException(
     super.message, {
-    this.lambdaName,
     super.recoverySuggestion,
     super.underlyingException,
-  });
+  }) : _message = message;
+
+  final String _message;
+
+  @override
+  String get message {
+    final match = _errorRegex.firstMatch(_message);
+    final parsedMessage = match?.group(2);
+    return parsedMessage ?? _message;
+  }
+
+  /// The name of the lambda which triggered this exception.
+  String? get lambdaName {
+    final match = _errorRegex.firstMatch(_message);
+    final lambdaName = match?.group(1);
```

Comment:
Same here

Reply:
No. There is no logical fallback. This is consistent with the existing behavior. `lambdaName` is nullable and has no fallback.